### PR TITLE
Misc bug fixes

### PR DESCRIPTION
**Pull request by Atlante45**
This PR fixes a collection of bugs I found with a static analyzer.

### CODE REVIEW

File: assignment-client/src/audio/AudioMixerSlave.cpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ bool AudioMixerSlave::prepareMix(const SharedNodePointer& listener) {
                 auto nodeID = node->getUUID();
 
                 // compute the node's max relative volume
-                float nodeVolume;
```
Comment (Atlante45) [inline assignment-client/src/audio/AudioMixerSlave.cpp]: `nodeVolume` is garbage data resulting in the final list of streams being incorrectly and randomly sorted every frame.

File: interface/src/AvatarBookmarks.cpp [hunk 2]
```diff
@@ -121,13 +121,13 @@ void AvatarBookmarks::addBookmark() {
...
-        bookmark->insert(ENTRY_AVATAR_URL, avatarUrl);
-        bookmark->insert(ENTRY_AVATAR_SCALE, avatarScale);
-        bookmark->insert(ENTRY_AVATAR_ATTACHMENTS, myAvatar->getAttachmentsVariant());
+        QVariantMap bookmark;
+        bookmark.insert(ENTRY_VERSION, AVATAR_BOOKMARK_VERSION);
+        bookmark.insert(ENTRY_AVATAR_URL, avatarUrl);
+        bookmark.insert(ENTRY_AVATAR_SCALE, avatarScale);
+        bookmark.insert(ENTRY_AVATAR_ATTACHMENTS, myAvatar->getAttachmentsVariant());
 
-        Bookmarks::addBookmarkToFile(bookmarkName, *bookmark);
```
Comment (Atlante45) [inline interface/src/AvatarBookmarks.cpp]: Memory leak

File: interface/src/AvatarEntitiesBookmarks.cpp [hunk 3]
```diff
@@ -144,13 +144,13 @@ void AvatarEntitiesBookmarks::addBookmark() {
...
-        bookmark->insert(ENTRY_AVATAR_URL, avatarUrl);
-        bookmark->insert(ENTRY_AVATAR_SCALE, avatarScale);
-        bookmark->insert(ENTRY_AVATAR_ENTITIES, myAvatar->getAvatarEntitiesVariant());
+        QVariantMap bookmark;
+        bookmark.insert(ENTRY_VERSION, AVATAR_ENTITIES_BOOKMARK_VERSION);
+        bookmark.insert(ENTRY_AVATAR_URL, avatarUrl);
+        bookmark.insert(ENTRY_AVATAR_SCALE, avatarScale);
+        bookmark.insert(ENTRY_AVATAR_ENTITIES, myAvatar->getAvatarEntitiesVariant());
 
-        Bookmarks::addBookmarkToFile(bookmarkName, *bookmark);
```
Comment (Atlante45) [inline interface/src/AvatarEntitiesBookmarks.cpp]: Memory leak

File: libraries/animation/src/Rig.cpp [hunk 4]
```diff
@@ -1244,7 +1244,7 @@ void Rig::updateHands(bool leftHandEnabled, bool rightHandEnabled, bool hipsEnab
         int handJointIndex = _animSkeleton->nameToJointIndex("LeftHand");
         int armJointIndex = _animSkeleton->nameToJointIndex("LeftArm");
         int elbowJointIndex = _animSkeleton->nameToJointIndex("LeftForeArm");
-        if (!leftArmEnabled && elbowJointIndex >= 0 && armJointIndex >= 0 && elbowJointIndex >= 0) {
```
Comment (Atlante45) [inline libraries/animation/src/Rig.cpp]: Copy/paste error

File: libraries/animation/src/Rig.cpp [hunk 5]
```diff
@@ -1291,7 +1291,7 @@ void Rig::updateHands(bool leftHandEnabled, bool rightHandEnabled, bool hipsEnab
         int handJointIndex = _animSkeleton->nameToJointIndex("RightHand");
         int armJointIndex = _animSkeleton->nameToJointIndex("RightArm");
         int elbowJointIndex = _animSkeleton->nameToJointIndex("RightForeArm");
-        if (!rightArmEnabled && elbowJointIndex >= 0 && armJointIndex >= 0 && elbowJointIndex >= 0) {
```
Comment (Atlante45) [inline libraries/animation/src/Rig.cpp]: Copy/paste error

File: libraries/networking/src/AddressManager.cpp [hunk 6]
```diff
@@ -666,7 +666,7 @@ bool AddressManager::handleViewpoint(const QString& viewpointString, bool should
             }
             
             emit locationChangeRequired(newPosition, orientationChanged, 
-                LookupTrigger::VisitUserFromPAL ? cancelOutRollAndPitch(newOrientation): newOrientation,
```
Comment (Atlante45) [inline libraries/networking/src/AddressManager.cpp]: Bad conditional

File: libraries/networking/src/NodeList.cpp [hunk 7]
```diff
@@ -999,7 +999,7 @@ void NodeList::setAvatarGain(const QUuid& nodeID, float gain) {
             }
 
             sendPacket(std::move(setAvatarGainPacket), *audioMixer);
-            QWriteLocker{ &_avatarGainMapLock };
```
Comment (Atlante45) [inline libraries/networking/src/NodeList.cpp]: Lock instantly goes out of scope, memory is unprotected.

File: libraries/networking/src/NodeList.cpp [hunk 8]
```diff
@@ -1011,7 +1011,7 @@ void NodeList::setAvatarGain(const QUuid& nodeID, float gain) {
 }
 
 float NodeList::getAvatarGain(const QUuid& nodeID) {
-    QReadLocker{ &_avatarGainMapLock };
```
Comment (Atlante45) [inline libraries/networking/src/NodeList.cpp]: Lock instantly goes out of scope, memory is unprotected.

File: libraries/plugins/src/plugins/PluginManager.cpp [hunk 9]
```diff
@@ -214,9 +214,10 @@ const DisplayPluginList& PluginManager::getDisplayPlugins() {
 }
 
 void PluginManager::disableDisplayPlugin(const QString& name) {
-    std::remove_if(_displayPlugins.begin(), _displayPlugins.end(), [&](const DisplayPluginPointer& plugin){
```
Comment (Atlante45) [inline libraries/plugins/src/plugins/PluginManager.cpp]: `std::remove_if` only return an iterator to what is to be deleted.

File: libraries/recording/src/recording/Deck.cpp [hunk 10]
```diff
@@ -187,14 +187,14 @@ void Deck::processFrames() {
 
 void Deck::removeClip(const ClipConstPointer& clip) {
     Locker lock(_mutex);
-    std::remove_if(_clips.begin(), _clips.end(), [&](const Clip::ConstPointer& testClip)->bool {
```
Comment (Atlante45) [inline libraries/recording/src/recording/Deck.cpp]: `std::remove_if` only return an iterator to what is to be deleted.

File: libraries/recording/src/recording/Deck.cpp [hunk 11]
```diff
@@ -187,14 +187,14 @@ void Deck::processFrames() {
...
+    _clips.remove_if([&](const Clip::ConstPointer& testClip)->bool {
         return (clip == testClip);
     });
 }
 
 void Deck::removeClip(const QString& clipName) {
     Locker lock(_mutex);
-    std::remove_if(_clips.begin(), _clips.end(), [&](const Clip::ConstPointer& clip)->bool {
```
Comment (Atlante45) [inline libraries/recording/src/recording/Deck.cpp]: `std::remove_if` only return an iterator to what is to be deleted.

File: libraries/shared/src/QVariantGLM.cpp [hunk 12]
```diff
@@ -66,7 +66,7 @@ void qListtoRgbColor(const QVariant& q, rgbColor& returnValue) {
 
 glm::vec3 qMapToGlmVec3(const QVariant& q) {
     QVariantMap qMap = q.toMap();
-    if (qMap.contains("x") && qMap.contains("y") && qMap.contains("y")) {
```
Comment (Atlante45) [inline libraries/shared/src/QVariantGLM.cpp]: Copy/paste error

File: libraries/shared/src/TriangleSet.h [hunk 13]
```diff
@@ -85,7 +85,7 @@ class TriangleSet {
 protected:
 
     bool _isBalanced{ false };
-    TriangleOctreeCell _triangleOctree;
     std::vector<Triangle> _triangles;
+    TriangleOctreeCell _triangleOctree;
```
Comment (Atlante45) [inline libraries/shared/src/TriangleSet.h]: `_triangleOctree` relies on `_triangles` for its initialization.

File: plugins/oculus/src/OculusBaseDisplayPlugin.cpp [hunk 14]
```diff
@@ -41,7 +41,7 @@ bool OculusBaseDisplayPlugin::beginFrameRender(uint32_t frameIndex) {
     std::array<glm::mat4, 2> handPoses;
     // Make controller poses available to the presentation thread
     ovr_for_each_hand([&](ovrHandType hand) {
-        static const auto REQUIRED_HAND_STATUS = ovrStatus_OrientationTracked & ovrStatus_PositionTracked;
+        static const auto REQUIRED_HAND_STATUS = ovrStatus_OrientationTracked | ovrStatus_PositionTracked;
```
Comment (Atlante45) [inline plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: Bits are incorrectly masked to 0.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6894/app_12328_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6894/HighFidelity-Beta-PR12328-8ebb3e52d8be7cfc452462acd59af585c75726a3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6894/HighFidelity-Beta-PR12328-8ebb3e52d8be7cfc452462acd59af585c75726a3.exe

File: plugins/oculus/src/OculusBaseDisplayPlugin.cpp (hunk 14 above)
Comment (jherico) [reply plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: Hey!  this is probably the source of the always present oculus hand tracker models!

**Comment by ZappoMan** [comment]
what's a test plan for this?

**Comment by Atlante45** [comment]
@ZappoMan I was thinking about that over the weekend. Some of those are gonna be tough to QA (ie. locks), so I was thinking smoke testing the module it's in.
I'll write something up tomorrow.

File: plugins/oculus/src/OculusBaseDisplayPlugin.cpp (hunk 14 above)
Comment (Atlante45) [reply plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: How can I verify that bug is fixed?

**Comment by Atlante45** [comment]
## Test Plan:
- [PR12191](https://github.com/highfidelity/hifi/pull/12191#issuecomment-358489508) Bookmarks (make sure avatar models work too)
- [C174](https://highfidelity.testrail.net/index.php?/cases/view/174) Touch controllers + IK (Make sure to use an Oculus Rift)
- [C6925](https://highfidelity.testrail.net/index.php?/cases/view/6925) Jump to a user from the PAL
- [C6927](https://highfidelity.testrail.net/index.php?/cases/view/6927) Change the volume of other users in the PAL
- [C147](https://highfidelity.testrail.net/index.php?/cases/view/147) Audio mixer still works

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by conklin94122** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7139/app_12328_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7139/HighFidelity-Beta-PR12328-8ebb3e52d8be7cfc452462acd59af585c75726a3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7139/HighFidelity-Beta-PR12328-8ebb3e52d8be7cfc452462acd59af585c75726a3.exe

**Comment by sbennett77** [comment]
Passed.

@conklin94122

**Pull request merged**